apollo-gradle-plugin/src/main/kotlin/com/apollographql/apollo3/gradle/internal/SchemaHelper.kt

```diff
@@ -49,10 +63,65 @@ internal object SchemaHelper {
 
     return response
   }
+
   /**
    * @param variables a map representing the variable as Json values
    */
-  internal fun executeQuery(query: String, variables: Map<String, Any>, url: String, headers: Map<String, String>): Response {
-    return executeQuery(mapOf("query" to query, "variables" to variables), url, headers)
+  internal fun executeQuery(
+      query: String,
+      variables: Map<String, Any>,
+      url: String,
+      headers: Map<String, String>,
+      insecure: Boolean = false,
+  ): Response {
+    return executeQuery(mapOf("query" to query, "variables" to variables), url, headers, insecure)
+  }
+
+  private fun OkHttpClient.Builder.applyInsecureTrustManager() = apply {
+    val insecureTrustManager = InsecureTrustManager()
+    sslSocketFactory(createSslSocketFactory(insecureTrustManager), insecureTrustManager)
+    hostnameVerifier(InsecureHostnameVerifier())
+  }
+
+  @Suppress("TooGenericExceptionCaught")
+  private fun createSslSocketFactory(trustManager: X509TrustManager): SSLSocketFactory {
+    try {
+      val sslContext = try {
+        SSLContext.getInstance("SSL")
+      } catch (_: Exception) {
+        Platform.get().newSSLContext()
```

Comment:
good question, but I don't remember why this workaround was used in our internal code. Under the hood it just calls `getInstance` with `"TLS"` argument. I think I'll remove this try catch block and leave only `SSLContext.getInstance("SSL")` call

Reply:
Although it depends on the platform cause method is `open` [see](https://github.com/square/okhttp/blob/okhttp_4.10.x/okhttp/src/main/kotlin/okhttp3/internal/platform/Platform.kt#L75)

Reply:
I found this page that gives **a lot** of detail about the differences. SSL is hard 😅  https://serverfault.com/questions/64484/functional-implications-of-differences-in-ssl-and-tls/368574#368574

SInce this is used on the insecure path and that's a lot of information, let's keep it like this. Do you mind adding a small comment like this:

```
// get a SSLContext.
// There are a lot of subtle differences in the different protocols but this is used on the insecure path
// we are ok taking any of them
```

So our future selves know where to dive if they want to?

Reply:
good idea